tests/specs/geoQuery.spec.js

```diff
@@ -1376,5 +1376,29 @@ describe("GeoQuery Tests:", function() {
         cl.x("p5");
       }).catch(failTestOnCaughtError);
     });
+
+    it("Calling cancel() in the middle of firing \"key_entered\" events is allowed", function(done) {
+      var cl = new Checklist(["p1", "cancel query", "loc1 entered"], expect, done);
+
+      geoQueries.push(geoFire.query({center: [1,2], radius: 1000}));
+
+      geoFire.set({
+        "loc1": [1, 2],
+        "loc2": [1, 3],
+        "loc3": [1, 4]
+      }).then(function() {
+        cl.x("p1");
+
+        var numKeyEnteredEventsFired = 0;
+        geoQueries[0].on("key_entered", function(key, location, distance) {
+          cl.x(key + " entered");
+          numKeyEnteredEventsFired++;
+          if (numKeyEnteredEventsFired === 1) {
+            cl.x("cancel query");
+            geoQueries[0].cancel();
```

Comment:
well it depends, since we will call done once the checklist is done, it will fail only if we have not finished running the tests yet, which might not be the case (since we explicitly called done for this test case).